adam/learner/learner_utils.py

```diff
@@ -191,3 +195,37 @@ def sort_by_num_nodes(g: DiGraph) -> int:
         computed_graph = graph
 
     return PerceptionGraphPattern(computed_graph, dynamic=perception_graph.dynamic)
+
+
+def covers_entire_utterance(
+    bound_surface_template: SurfaceTemplateBoundToSemanticNodes,
+    language_concept_alignment: LanguageConceptAlignment,
+    *,
+    ignore_determiners: bool = False,
+) -> bool:
+    num_covered_tokens = 0
+    for element in bound_surface_template.surface_template.elements:
+        if isinstance(element, str):
+            num_covered_tokens += 1
+        else:
+            num_covered_tokens += len(
+                language_concept_alignment.node_to_language_span[
+                    bound_surface_template.slot_to_semantic_node[element]
+                ]
+            )
+    # We may need to
+    sized_tokens = (
+        len(language_concept_alignment.language.as_token_sequence())
+        if not ignore_determiners
+        else len(
+            [
+                token
+                for token in language_concept_alignment.language.as_token_sequence()
+                if token not in ["a", "the"]
```

Comment:
Do we have a "constant" for the determiner set somewhere?  I think we also somewhere track English-specific places in the code (e.g. on an issue). If so, add this there.

Reply:
Added: https://github.com/isi-vista/adam/issues/498#issuecomment-654942842